src/UI/WB.UI.Designer/Services/Restore/QuestionnaireRestoreService.cs

```diff
@@ -72,22 +67,31 @@ public Guid RestoreQuestionnaire(Stream archive, Guid responsibleId, RestoreStat
             return questionnaire.PublicKey;
         }
 
-        public QuestionnaireDocument RestoreQuestionnaireFromZipFileOrThrow(ZipInputStream zipStream, Guid responsibleId,
+        public QuestionnaireDocument RestoreQuestionnaireFromZipFileOrThrow(ZipArchive zipStream, Guid responsibleId,
             RestoreState state, bool createNew)
         {
-            ZipEntry zipEntry;
-            while ((zipEntry = zipStream.GetNextEntry()) != null)
+            foreach (var zipEntry in zipStream.Entries)
             {
-                if (zipEntry.IsDirectory) continue;
+                if (zipEntry.FullName.EndsWith("/")) continue;
+                
                 try
                 {
-                    string[] zipEntryPathChunks = zipEntry.FileName.Split(
+                    string[] zipEntryPathChunks = zipEntry.FullName.Split(
                         new[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar },
                         StringSplitOptions.RemoveEmptyEntries);
 
                     if (zipEntryPathChunks.Length == 1 && zipEntryPathChunks[0].ToLower().Equals("document.json"))
                     {
-                        string textContent = new StreamReader(zipStream, Encoding.UTF8).ReadToEnd();
+                        byte[] binaryContent;
+                        using (var memoryStream = new MemoryStream())
+                        {
+                            zipEntry.Open().CopyTo(memoryStream);
+                            binaryContent = memoryStream.ToArray();
+                        }
+                        
+                        string textContent = Encoding.UTF8.GetString(binaryContent);
```

Comment:
move to extensions for ZipEntity.
like zipEntity.GetString() and zipEntity.GetBytess()
And call this extension in anather places